### PR TITLE
Cli/hashtable

### PR DESCRIPTION
**Pull request by Gestalt-R**
Merging basic hashtable functionality. Detailed description can be found in issue #103.

### CODE REVIEW

File: src/cli/include/cmd.h [hunk 1]
```diff
@@ -1,21 +1,112 @@
...
 #include "parser.h"
+//DELETETHISPLACEHOLDER, it has a meaningless definition of game_t that is not implemented yet
+#include "placeholder.h"
 #include "operations.h"
+#include "../../common/include/uthash.h"
+//#include "../../game-state/include/game.h"
 
 /* Operation data type */
-typedef char *operation(char *tokens[TOKEN_LIST_SIZE]);
+typedef char * operation(char *tokens[TOKEN_LIST_SIZE], game_t * game);
```
Comment (Tianlel) [inline src/cli/include/cmd.h]: Just a minor style problem, Elizabeth has already commented on this but you haven't changed how these pointers-- "Everywhere you have a pointer, make it like "char *name" not "char * name" "

File: src/cli/include/operations.h [hunk 2]
```diff
@@ -27,51 +28,51 @@
...
- * and pass a command along.   
+/* These functions will generate an action-struct (based on action management)
+ * and pass a command along.
  *
  * Parameters:
  * - tokens: parsed input string (validified)
- *
+ *  - pointer to game to be altered, unused
  * Returns:
  * - Said list of supported operations as a string
```
Comment (Tianlel) [inline src/cli/include/operations.h]: Shouldn't these return some system message indicating whether an error has occured or how the game state has changed?

File: src/cli/src/cmd.c [hunk 3]
```diff
@@ -6,6 +6,61 @@
 #include "cmd.h"
 #include "validate.h"
 
+/* === hashtable constructors === */
+
+void add_entry(char * command_name, operation * associated_operation, lookup_t * * table){
```
Comment (Tianlel) [inline src/cli/src/cmd.c]: We have been following the Allman style, so could you modify your code accordingly to keep things consistent? Also indentation should be 4 spaces not 2

Comment (Gestalt-R) [reply src/cli/src/cmd.c]: Done.

File: src/cli/include/operations.h (hunk 2 above)
Comment (Gestalt-R) [reply src/cli/include/operations.h]: I'm not implementing them. This was unchanged, except I added a change I made to the action struct. The description is horribly inaccurate though. It seems to be made for a completely different function than anything we had planned at any point.

The answer is yes, but the error was already there.

File: src/cli/include/cmd.h (hunk 1 above)
Comment (Gestalt-R) [reply src/cli/include/cmd.h]: Fixed with Astyle

File: src/cli/src/cmd.c (hunk 3 above)
Comment (Gestalt-R) [reply src/cli/src/cmd.c]: Astyle.

**Comment by lydiafilipe** [comment]
Issue Score: ✓-

Comments:
Good work on the comments and issue referencing, however the title is vague and there is no closing statement.

**Comment by lydiafilipe** [comment]
Issue Score: ✓-

Comments:
Good work on the comments and issue referencing, however the title is vague and there is no closing statement.

**Pull request merged**